changelog/0087-upgrade-to-es-6.yml

```
@@ -0,0 +1,409 @@
+      cchq <env> django-manage elastic_sync_multiplexed remove_residual_indices
+    ```
+
+  9. At this state you would be ready to upgrade the Elasticsearch version.
+
+  ## Upgrade the Elasticsearch version
+
+    The steps below outline the process to upgrade the Elasticsearch version from 5.6.16 to 6.8.23.
+  
+  ### Pre-Upgrade Steps
+
+    1. Start by ssh-ing into the control machine...
+    ```sh
+      cchq <env> ssh control
+    ```
+
+    2. Start a new tmux session
+    ```sh
+      tmux new -s es_upgrade
+    ```
+
+    3. For commands including `${ENV}` and `${ES_HOST}` below to work, make sure to set `ENV` to the name of your environment, and `ES_HOST` to the IP address of one of your ES nodes:
+    ```sh
+      ENV=<env>
+      ES_HOST=$(cchq ${ENV} lookup elasticsearch:0 | cut -d' ' -f1)
+    ```
+    4. Make sure that the disk usage is below 47%,
+  or that you have a place to back up to on another disk or remotely. You can check disk usage on each elasticsearch host usage by running:
+    ```sh
+      cchq ${ENV} run-shell-command elasticsearch  "du -sh  /opt/data/elasticsearch-5.6.16" -b
+    ```
+
+    5. After you have figured out the backup strategy and ensured sufficient disk space, continue with the steps below.
+  
+  ### Upgrade Steps
+
+  1. Add the following parameters to your environment's `public.yml`:
+    
+    ```yaml
+    elasticsearch_version: 6.8.23
+    elasticsearch_download_sha256: 424af91f838f9e5f13e0292f97cbd6333535450291a621d761bd479dfc2dff78.
+    .
+    .
+    localsettings:
+      ...
+      ELASTICSEARCH_MAJOR_VERSION: 6
+    ```
+
+  2. Pull the latest commcare-cloud changes on control machine.
+    
+    ```sh
+      cd ${COMMCARE_CLOUD_REPO} && git pull
+    ```
+
+  3. Stop commcare services.
+
+    ```sh
+      cchq ${ENV} downtime start
+    ```
+  
+  4. Apply the settings that you have updated in Step 1.
+
+    ```sh
+      cchq ${ENV} update-config
+    ```
+
+  5. Disable Elasticsearch shard allocation. It should already be disabled, but run this command to ensure that it is already disabled.
+
+    ```sh
+      curl -X PUT "${ES_HOST}:9200/_cluster/settings?pretty" -H 'Content-Type: application/json' -d'
+      {
+        "persistent": {
+          "cluster.routing.allocation.enable": "none"
+        }
+      }
+      '
+
+    ```
+
+  6. Perform a synced flush.
+
+    ```sh
+      curl -X POST "${ES_HOST}:9200/_flush/synced?pretty"  # safe to reissue a few times if it fails
+    ```
+
+  7. Stop the Monit and Elasticsearch services.
+
+    ```sh
+      cchq ${ENV} run-shell-command elasticsearch "service monit stop" -b
+
+      cchq ${ENV} run-shell-command elasticsearch "service elasticsearch stop" -b
+    ```
+
+  8. Make a backup of the 5.6.16 data directory. If you are backing up on the same machine then one way to do it is by:
+    
+    ```sh
+      cchq ${ENV} run-shell-command elasticsearch "rsync -av /opt/data/elasticsearch-5.6.16 /opt/data/elasticsearch-5.6.16-backup" -b
```

This will create a new directory at `/opt/data/elasticsearch-5.6.16-backup/elasticsearch-5.6.16`, but adding a slash to the end of each path should fix it. The directory `/opt/data/elasticsearch-5.6.16-backup` will be created if it does not exist.

```suggestion
      cchq ${ENV} run-shell-command elasticsearch "rsync -av /opt/data/elasticsearch-5.6.16/ /opt/data/elasticsearch-5.6.16-backup/" -b
```